src/dfx/src/lib/environment.rs

```diff
@@ -358,7 +375,16 @@ impl<'a> Environment for AgentEnvironment<'a> {
     }
 
     fn get_effective_canister_id(&self) -> Principal {
-        self.backend.get_effective_canister_id()
+        self.backend
+            .get_override_effective_canister_id()
+            .unwrap_or_else(|| {
+                self.effective_canister_id
```

Comment:
I wonder why this is attempted in the middle.

Reply:
It's more important than the underlying Environment's previous setting but not as important as the user override.